pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -91,6 +97,33 @@ func (hs *proxierHealthServer) QueuedUpdate() {
 	hs.oldestPendingQueued.CompareAndSwap(zeroTime, hs.clock.Now())
 }
 
+// IsHealthy returns the proxier's health state. The result reflects its
+// performance w.r.t updating the service rules on the node, specifically:
+// it's healthy when it's fully synced and not timing out syncing the
+// service rules.
+func (hs *proxierHealthServer) IsHealthy() (bool, time.Time, time.Time) {
```

Comment:
We are exporting a method that outputs two `time.Time` var that are never used and are really obscure.
I suggest to have an internal variable with the healthy status and use IsHealthy() to return its value only, keeping this logic where it was. Same as you did with the services_health
However, wait for @danwinship and @thockin opinions on this

Reply:
Yes, that makes more sense. I will change